src/uniprotkb/components/entry/GoCam.tsx

```diff
@@ -0,0 +1,162 @@
+import { ReactNode, useEffect, useMemo, useState } from 'react';
+import { ExternalLink, Loader } from 'franklin-sites';
+import pMap from 'p-map';
+
+import ErrorHandler from '../../../shared/components/error-pages/ErrorHandler';
+import GoCamViz from '../protein-data-views/GoCamViz';
+
+import { useSmallScreen } from '../../../shared/hooks/useMatchMedia';
+import useDataApi from '../../../shared/hooks/useDataApi';
+
+import fetchData from '../../../shared/utils/fetchData';
+import { heuristic } from '../../../tools/state/utils/heuristic';
+
+import externalUrls from '../../../shared/config/externalUrls';
+
+import { GoCamModelInfo, GoCamModels } from '../../types/goCamTypes';
+
+import styles from './styles/go-cam.module.scss';
+
+const extractGoCamId = (url: string) => {
+  const reGoCamId = /https?:\/\/model\.geneontology\.org\/(?<goCamId>.*)/;
+  const m = url.match(reGoCamId);
+  return m?.groups?.goCamId;
+};
+
+export const getGoCamStructures = (data: GoCamModels[] = []) => {
+  const idToItem = new Map<string, { id: string; label: string }>();
+  for (const d of data) {
+    const id = extractGoCamId(d.gocam);
+    if (id) {
+      idToItem.set(id, {
+        id,
+        label: d.title,
+      });
+    }
+  }
+  return idToItem;
+};
+
+export const isUniprotCurated = (goCamModel: GoCamModelInfo) =>
+  goCamModel?.annotations?.some(
+    ({ key, value }) =>
+      key === 'providedBy' && value === 'https://www.uniprot.org'
+  );
+
+type Props = {
+  primaryAccession: string;
+};
+
+const GoCam = ({ primaryAccession }: Props) => {
+  const isSmallScreen = useSmallScreen();
+  const [selectedId, setSelectedId] = useState<string | null>(null);
+  const allGoCamIdsResponse = useDataApi<GoCamModels[]>(
+    isSmallScreen ? null : externalUrls.GeneOntologyModels(primaryAccession)
+  );
+  const [uniprotGoCamIds, setUniprotGoCamIds] = useState<string[] | null>(null);
+
+  const goCamIdToItem = useMemo(
+    () => getGoCamStructures(allGoCamIdsResponse.data),
+    [allGoCamIdsResponse.data]
+  );
+
+  useEffect(() => {
+    async function fetchGoCamModels() {
+      if (goCamIdToItem.size) {
+        const mapper = (id: string) =>
+          fetchData<GoCamModelInfo>(
+            externalUrls.GeneOntologyModelInfo(id)
+          ).then((response) => ({
+            id,
+            data: response.data,
+          }));
+        const results = await pMap(Array.from(goCamIdToItem.keys()), mapper, {
+          concurrency: heuristic.concurrency,
+        });
+        setUniprotGoCamIds(
+          results
+            .filter(({ data }) => isUniprotCurated(data))
+            .map(({ id }) => id)
+        );
+      }
+    }
+    fetchGoCamModels();
+  }, [goCamIdToItem]);
+
+  useEffect(() => {
+    if (uniprotGoCamIds?.[0]) {
+      setSelectedId(uniprotGoCamIds[0]);
+    }
+  }, [uniprotGoCamIds]);
```

Comment:
This bit of logic might be able to live directly in the previous useEffect, at the same time as `setUniprotGoCamIds`, to avoid one re-render. Would require using `useSafeState` for `selectedId` though